pkg/clusters/addons/kuma/addon.go

```diff
@@ -63,7 +63,10 @@ func (a *Addon) Namespace() string {
 
 // Version indicates the Kuma version for this addon.
 func (a *Addon) Version() semver.Version {
-	return a.version
+	if a.version == nil {
+		return semver.Version{}
+	}
+	return *a.version
```

Comment:
The downside of this change is that you'll get a `v0.0.0` even if you didn't set a version, so the caller would need to guard for this.

It's been a while since I dug around in this code, but couldn't we get away with changing the signature for this function instead?

```suggestion
func (a *Addon) Version() (semver.Version, error) {
```

And return an error when not set? :thinking:

Reply:
Yup. Technically that is a breaking change, but given we're still before v1 we have an excuse I guess. 👍 4963e7b